lib/ansible/modules/blockinfile.py

```diff
@@ -269,7 +269,7 @@ def main():
             module.fail_json(rc=257,
                              msg='Path %s does not exist !' % path)
         destpath = os.path.dirname(path)
-        if not os.path.exists(destpath) and not module.check_mode:
+        if destpath and not os.path.exists(destpath) and not module.check_mode:
             try:
                 os.makedirs(destpath)
             except Exception as e:
```

Comment:
Addressing this in this PR isn't required, but this subscripting of the exception only works in Python 2, and as such an exception here would cause the exception seen in the report on Python3 while building the message for `fail_json`.

Reply:
I created #81662 for that.

Reply:
#81662 has been merged.